os/kernel/binary_manager/binary_manager_load.c

```diff
@@ -405,127 +355,175 @@ static int binary_manager_reload(int bin_idx)
-	switch (load_cmd) {
-	case LOADCMD_LOAD_ALL:
-		ret = binary_manager_load_all();
-		break;
-	case LOADCMD_UPDATE:
-		if (argc <= 2) {
-			bmdbg("Invalid arguments for reloading, argc %d\n", argc);
-			break;
-		}
-		/* [2] binary name for reloading */
-		ret = binary_manager_update(argv[2]);
-		break;
-#ifdef CONFIG_BINMGR_RECOVERY
-	case LOADCMD_RELOAD:
-		if (argc <= 2) {
-			bmdbg("Invalid arguments for reloading, argc %d\n", argc);
-			break;
-		}
-		/* [2] binary id for reloading */
-		int bin_idx = (int)atoi(argv[2]);
+	/* [1] binary index for reloading */
+	int bin_idx = (int)atoi(argv[1]);
 #ifdef CONFIG_SUPPORT_COMMON_BINARY
-		if (bin_idx == BM_BINID_LIBRARY) {
-			/* Reload common library and all binaries */
-			char libname[CONFIG_NAME_MAX];
-			snprintf(libname, CONFIG_NAME_MAX, "%s%s", CONFIG_COMMON_BINARY_PATH, CONFIG_COMMON_BINARY_NAME);
-			ret = load_binary(COMMLIB_IDX, libname, NULL);
-			if (ret < 0) {
-				return BINMGR_OPERATION_FAIL;
-			}
+	if (bin_idx == BM_BINID_LIBRARY) {
+		/* Reload common library and all binaries */
+		char libname[CONFIG_NAME_MAX];
+		snprintf(libname, CONFIG_NAME_MAX, "%s%s", CONFIG_COMMON_BINARY_PATH, CONFIG_COMMON_BINARY_NAME);
+		ret = load_binary(COMMLIB_IDX, libname, NULL);
+		if (ret < 0) {
+			return BINMGR_OPERATION_FAIL;
+		}
 
-			int bidx;
-			int bin_count = binary_manager_get_ucount();
+		int bidx;
+		int bin_count = binary_manager_get_ucount();
 
-			for (bidx = 1; bidx <= bin_count; bidx++) {
-				ret = binary_manager_reload(bidx);
-				if (ret < 0) {
-					return ret;
-				}
+		for (bidx = 1; bidx <= bin_count; bidx++) {
+			ret = binary_manager_reload(bidx);
+			if (ret < 0) {
+				return ret;
 			}
-		} else
+		}
+	} else
 #endif
-			ret = binary_manager_reload(bin_idx);
-		break;
+	return binary_manager_reload(bin_idx);
+}
 #endif
-	default:
-		bmdbg("Invalid loading command %d\n", load_cmd);
+
+/****************************************************************************
+ * Name: update_thread
+ *
+ * Description:
+ *   This thread updates binary with binary index.
+ *   It terminates all the task/thread created by the binary and loads the binary.
+ *
+ ****************************************************************************/
+static int update_thread(int argc, char *argv[])
+{
+	int ret;
+	int bin_idx;
+
+	if (argc <= 1) {
+		bmdbg("Invalid arguments for update, argc %d\n", argc);
+		return ERROR;
 	}
-	bmvdbg("Loading result %d\n", ret);
 
-	return ret;
+	/* [1] binary index for update */
+	bin_idx = (int)atoi(argv[1]);
+	if (bin_idx < 0 || bin_idx > USER_BIN_COUNT) {
+		bmdbg("Invalid index %d\n", bin_idx);
+		return ERROR;
+	}
+
+	if (BIN_STATE(bin_idx) < BINARY_INACTIVE || BIN_STATE(bin_idx) > BINARY_RUNNING) {
+		bmdbg("Invalid binary state %d\n", BIN_STATE(bin_idx));
+		return BINMGR_OPERATION_FAIL;
+	}
+
+	/* Terminate binary if binary is already loaded */
+	if (BIN_STATE(bin_idx) == BINARY_LOADING_DONE || BIN_STATE(bin_idx) == BINARY_RUNNING) {
```

Comment:
If there is no new binary, so we don't need to update, then we don't need to terminate the previous one.
We can just ignore the update request.
Please check this logic.

Reply:
yes. I agree with your opinion. I will modify later.